### PR TITLE
add WithLocalAddr option 

### PR DESCRIPTION
**Pull request by ta3pks**
Ability to bind to a specific local address

### CODE REVIEW

**Comment by bogdanfinn** [comment]
@NikosEfthias thank you for the PR. In general it looks good to me, but can you maybe give me an example why this option is needed? I mean describe a use case or something like that? 

I'm currently not getting it 😆

**Comment by NikosEfthias** [comment]
This is needed when one has multiple IP addresses assigned to the same machine.
Currently, there is no way to select from which source IP address to send request from, so that's why I created this pr

File: connect.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ type connectDialer struct {
 // newConnectDialer creates a dialer to issue CONNECT requests and tunnel traffic via HTTP/S proxy.
 // proxyUrlStr must provide Scheme and Host, may provide credentials and port.
 // Example: https://username:password@golang.org:443
-func newConnectDialer(proxyUrlStr string, timeout time.Duration) (proxy.ContextDialer, error) {
+func newConnectDialer(proxyUrlStr string, timeout time.Duration, localAddr *net.TCPAddr) (proxy.ContextDialer, error) {
```
Comment (bogdanfinn) [inline connect.go:92]: i think we want this also on `func newDirectDialer(timeout time.Duration) proxy.ContextDialer` which is used when no proxyUrl is used.

Comment (bogdanfinn) [reply connect.go:92]: @NikosEfthias got it thank you. Then i assume we want this option also on the directDialer (and maybe even von the Socks dialer?). Currently your implementation sets this option only on the proxy dialer

**Comment by NikosEfthias** [comment]
just added it for all the types

**Comment by ta3pks** [comment]
@bogdanfinn is there any roadblocks preventing this from being merged at the moment?

**Pull request merged**